x/concentrated-liquidity/incentives_test.go

```diff
@@ -969,83 +1053,37 @@ func (s *KeeperTestSuite) TestUpdateUptimeAccumulatorsToNow() {
-
-			if tc.expectedError != nil {
-				s.Require().Error(err)
-				s.Require().ErrorContains(err, tc.expectedError.Error())
+			// System under test 1
+			// Use cache context to avoud persisting updates for the next function
+			// that relies on the same test cases and setup.
+			cacheCtx, _ := s.Ctx.CacheContext()
+			err = clKeeper.UpdateUptimeAccumulatorsToNow(cacheCtx, tc.poolId)
 
-				// Ensure accumulators remain unchanged
-				newUptimeAccumValues, err := clKeeper.GetUptimeAccumulatorValues(s.Ctx, clPool.GetId())
-				s.Require().NoError(err)
-				s.Require().Equal(initUptimeAccumValues, newUptimeAccumValues)
+			validateResult(cacheCtx, err, tc, balancerPoolId, clPool.GetId(), initUptimeAccumValues, qualifyingBalancerLiquidity, qualifyingLiquidity)
 
-				// Ensure incentive records remain unchanged
-				updatedIncentiveRecords, err := clKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPool.GetId())
-				s.Require().NoError(err)
-				s.Require().Equal(tc.poolIncentiveRecords, updatedIncentiveRecords)
+			// System under test 2
 
+			// Skip this test case as UpdatePoolGivenUptimeAccumulatorsToNow relies
+			// on this check to be done by the caller.
+			if errors.Is(tc.expectedError, types.PoolNotFoundError{PoolId: invalidPoolId}) {
 				return
 			}
 
+			uptimeAccs, err := clKeeper.GetUptimeAccumulators(s.Ctx, tc.poolId)
 			s.Require().NoError(err)
 
-			// Get updated pool for testing purposes
-			clPool, err = clKeeper.GetPoolById(s.Ctx, tc.poolId)
-			s.Require().NoError(err)
-
-			// Calculate expected uptime deltas using qualifying liquidity deltas.
-			// Recall that uptime accumulators track emitted incentives / qualifying liquidity.
-			expectedUptimeDeltas := []sdk.DecCoins{}
-			for _, curSupportedUptime := range types.SupportedUptimes {
-				// Calculate expected incentives for the current uptime by emitting incentives from
-				// all incentive records to their respective uptime accumulators in the pool.
-				// TODO: find a cleaner way to calculate this that does not involve iterating over all incentive records for each uptime accum.
-				curUptimeAccruedIncentives := cl.EmptyCoins
-				for _, poolRecord := range tc.poolIncentiveRecords {
-					if poolRecord.MinUptime == curSupportedUptime {
-						// We set the expected accrued incentives based on the total time that has elapsed since position creation
-						curUptimeAccruedIncentives = curUptimeAccruedIncentives.Add(sdk.NewDecCoins(expectedIncentivesFromRate(poolRecord.IncentiveDenom, poolRecord.IncentiveRecordBody.EmissionRate, defaultTestUptime+tc.timeElapsed, qualifyingLiquidity))...)
-					}
-				}
-				expectedUptimeDeltas = append(expectedUptimeDeltas, curUptimeAccruedIncentives)
-			}
-
-			// Get new uptime accum values for comparison
-			newUptimeAccumValues, err := clKeeper.GetUptimeAccumulatorValues(s.Ctx, tc.poolId)
-			s.Require().NoError(err)
+			err = clKeeper.UpdateGivenPoolUptimeAccumulatorsToNow(s.Ctx, clPool, uptimeAccs)
```

Comment:
Can you add more comments about the reason upon calling `updateGivenPoolUptimeAccumulatorsToNow` here again and what we're trying to achieve? Also on a side note, thoughts on having this as a separate unit test?

Reply:
done